commit a72021ae99021409319d552dad744572b20febfb
Author: Noralf Trønnes <noralf@tronnes.org>
Date:   Sat Aug 15 20:50:02 2015 +0200

    bcm2835: Add support for uart1
    
    This is a hack until a proper solution is agreed upon.
    Martin Sperl is doing some work in this area.
    
    Signed-off-by: Noralf Trønnes <noralf@tronnes.org>

diff --git a/arch/arm/mach-bcm/board_bcm2835.c b/arch/arm/mach-bcm/board_bcm2835.c
index 1e6f1cf..ea36eec 100644
--- a/arch/arm/mach-bcm/board_bcm2835.c
+++ b/arch/arm/mach-bcm/board_bcm2835.c
@@ -22,6 +22,29 @@
 #include <asm/mach/arch.h>
 #include <asm/mach/map.h>
 
+/* Use this hack until a proper solution is agreed upon */
+static void __init bcm2835_init_uart1(void)
+{
+	struct device_node *np;
+
+	np = of_find_compatible_node(NULL, NULL, "brcm,bcm2835-aux-uart");
+	if (of_device_is_available(np)) {
+		np = of_find_compatible_node(NULL, NULL,
+					     "bcrm,bcm2835-aux-enable");
+		if (np) {
+			void __iomem *base = of_iomap(np, 0);
+
+			if (!base) {
+				pr_err("bcm2835: Failed enabling Mini UART\n");
+				return;
+			}
+
+			writel(1, base);
+			pr_info("bcm2835: Mini UART enabled\n");
+		}
+	}
+}
+
 static void __init bcm2835_init(void)
 {
 	struct device_node *np = of_find_node_by_path("/system");
@@ -42,6 +65,8 @@ static void __init bcm2835_init(void)
 		system_rev = val;
 	if (!of_property_read_u64(np, "linux,serial", &val64))
 		system_serial_low = val64;
+
+	bcm2835_init_uart1();
 }
 
 static const char * const bcm2835_compat[] = {
diff -uN a/1.txt b/1.txt
--- a/dummy/rpi_1024_a72021ae99021409319d552dad744572b20febfb.txt	1970-01-01 00:00:00.000000000 +0000
+++ b/dummy/rpi_1024_a72021ae99021409319d552dad744572b20febfb.txt	2013-12-23 04:07:40.000000000 +0000
@@ -0,0 +1 @@
+dummy file to ensure patch has content.
